internal/subscriber/subscriber.go

```diff
@@ -110,80 +110,20 @@ func (s *Subscriber) retry(ctx context.Context, msg messagebroker.ReceivedMessag
 		return
 	}
 
-	// commit on the primary topic after message has been submitted for retry
-	_, err = s.consumer.CommitByPartitionAndOffset(ctx, messagebroker.CommitOnTopicRequest{
-		Topic:     msg.Topic,
-		Partition: msg.Partition,
-		// add 1 to current offset
-		// https://docs.confluent.io/5.5.0/clients/confluent-kafka-go/index.html#pkg-overview
-		Offset: msg.Offset + 1,
-	})
-	if err != nil {
-		logger.Ctx(ctx).Errorw("subscriber: failed to commit message", "logFields", s.getLogFields(), "error", err.Error())
-		s.errChan <- err
-		return
-	}
+	s.commitAndRemoveFromMemory(ctx, msg)
```

Comment:
can we call acknowledge directly here ?

Reply:
Acknowledge:  

    1. Do some checks
    2. Commit continous offsets
    3. Remove from memory
    4. Update the metrics related to acks

Have extracted points 2 and 3 (the common part across ack, deadline, nack)